3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -0,0 +1,148 @@
+        baseCurrency  <- ScalazJson4sUtils.extract[String](config, "parameters", "baseCurrency")
+        rateAt        <- ScalazJson4sUtils.extract[String](config, "parameters", "rateAt")
+        enrich        =  CurrencyConversionEnrichment(apiKey, baseCurrency, rateAt)
+      } yield enrich).toValidationNel
+    })
+  }
+}
+
+// Object and a case object with the same name
+case class CurrencyConversionEnrichment(
+  apiKey: String,
+  bsCurrency: String, 
+  rateAt: String) extends Enrichment {
+
+  val version = new DefaultArtifactVersion("0.1.0")
+
+  // To provide Validation for org.joda.money.Money variable
+  def eitherToValidation(input:Either[OerResponseError, Money]): Validation[String, String]= {
+    input match {
+      case Right(l) => (l.getAmount().toPlainString()).success[String]
+      case Left(l) => (s"Open Exchange Rates error, message: ${l.errorMessage}").failure[String]
+    }
+  }
+
+  /**
+  * Convert's currency for a given
+  * set of currency, using
+  * Scala-Forex.
+  *
+  * @param trCurrency The desired
+  *        currency for a given 
+  *        amount
+  * @param trAmounts Contains
+  *        total amount, tax, shipping
+  *        amount's
+  * @param tiCurrency Trasaction Item
+  *        Currency
+   * @param tiPrice Trasaction Item
+  *         Price
+  * @return the converted currency
+  *         in TransacrionAmounts
+  *         format and Ttansaction
+  *         Item Price 
+  */
+  def convertCurrencies(trCurrency: Option[String], trAmountsTotal: Double, trAmountsTax: Double, trAmountsShipping: Double, tiCurrency: Option[String], tiPrice: Double): ValidationNel[String, (String, String, String, String)] = {
+    
+    try{
+        val fx = Forex(ForexConfig( nowishCacheSize = 0, nowishSecs = 0, eodCacheSize= 0, baseCurrency = bsCurrency), OerClientConfig(apiKey, DeveloperAccount))  
+        trCurrency match{
+          case Some(tr) => {
+            tiCurrency match {
+              case Some(ti) =>{ 
+                val newCurrencyTr  = fx.convert(trAmountsTotal, tr).to(bsCurrency).nowish
+                val newCurrencyTi = fx.convert(tiPrice, ti).to(bsCurrency).nowish 
+                val newTrAmountsTax = fx.convert(trAmountsTax, tr).to(bsCurrency).nowish
+                val newTrAmountsShipping = fx.convert(trAmountsShipping, tr).to(bsCurrency).nowish
+                //val transactionAmounts   = (eitherToValidation(newCurrencyTr).toValidationNel |@| eitherToValidation(newTrAmountsTax).toValidationNel |@| eitherToValidation(newTrAmountsShipping).toValidationNel ) { //e 
+                //                             (_, _, _)
+                //                           }
+                val transactionAmountsTuple = (eitherToValidation(newCurrencyTr).toValidationNel |@| eitherToValidation(newTrAmountsTax).toValidationNel |@| eitherToValidation(newTrAmountsShipping).toValidationNel |@| eitherToValidation(newCurrencyTi).toValidationNel) { //e 
```

Comment:
Either remove the unnecessary val assignment, or if that gives you a compile error (I don't see why it would), then add `// TODO: remove unnecessary val assignment`